### ISSUE TITLE
Load/destroy the maplibre map only when visible/hidden and  allow map.getFeaturesAtPixel

### ISSUE THREAD

**Issue by oterral** [geoblocks/ol-maplibre-layer#154]
Hi @fredj ,

Inspired by this project we have developed our own MaplibreLayer in [mobility-toolbox-js](https://github.com/geops/mobility-toolbox-js) ,for the [v3](https://github.com/geops/mobility-toolbox-js/tree/v3) we have switched to a complete [renderer](https://github.com/geops/mobility-toolbox-js/blob/v3/src/ol/renderers/MaplibreLayerRenderer.ts) class instead of the use of the render, it allows us to use the `map.getFeaturesAtPixel` function.

So the difference between ol-maplibre-layer  and mobility-toolbox-js:

- load the malibre map only when visible
- destroy the maplibre map when hidden
- automatically update attributions when source data changes
- relaod the maplibre map on target change event
- allow the use of `map.getFeaturesAtPixel ` to get vector tiles features as OpenLayer feature

Since I don't like duplicated code, we would be happy to propose a PR  for  those improvements.   Are you interested in ?

**Comment by oterral** [comment]
Maybe @ger-benjamin

**Comment by ger-benjamin** [comment]
Hello @oterral ,

Sorry, we don't really have/take time to manage a community. We easily miss messages.
But PRs are welcome :-)

What do you propose exactly ? Adding the features described above, in the aim to use this project and not anymore the matching code in mobility-toolbox ? As long as it's generically usable, yes that would be 100% in the spirit of geoblocks.

**Comment by oterral** [comment]
>  Adding the features described above, in the aim to use this project and not anymore the matching code in mobility-toolbox ? As long as it's generically usable, yes that would be 100% in the spirit of geoblocks.

Yes that's exactly the spirit. We use the maplibre layer very often and sometimes we don't want to have to import mobility-toolbox-js in all our project if it's not needed. Since it's very generic, geoblocks seemed to be a good place.

I will prepare a PR
